java/util/src/test/java/com/google/protobuf/util/JsonFormatTest.java

```diff
@@ -1173,7 +1173,41 @@ public void testParserIgnoringUnknownFields() throws Exception {
     String json = "{\n" + "  \"unknownField\": \"XXX\"\n" + "}";
     JsonFormat.parser().ignoringUnknownFields().merge(json, builder);
   }
+  
+  
+  public void testParserIgnoringUnknownEnums() throws Exception {
+    TestAllTypes.Builder builder = TestAllTypes.newBuilder();
+    String json = "{\n" + "  \"optionalNestedEnum\": \"XXX\"\n" + "}";
+    JsonFormat.parser().ignoringUnknownFields().merge(json, builder);
+    assertEquals(0, builder.getOptionalNestedEnumValue());
+  }
+
+  public void testUnknownEnumMap() throws Exception {
+    TestMap.Builder builder = TestMap.newBuilder();
+    JsonFormat.parser().ignoringUnknownFields().merge(
+        "{\n"
+          + "  \"int32ToEnumMap\": {1: XXX, 2: FOO}"
+          + "}",
+        builder);
 
+    assertEquals(Integer.valueOf(0), builder.getInt32ToEnumMapValueMap().get(1));
+    assertEquals(NestedEnum.FOO, builder.getInt32ToEnumMapMap().get(2));
+  }
+  
+  public void testRepeatedUnknownEnum() throws Exception {
+    TestAllTypes.Builder builder = TestAllTypes.newBuilder();
+    JsonFormat.parser().ignoringUnknownFields().merge(
+        "{\n"
+          + "  \"repeatedNestedEnum\": [XXX, FOO, BAR, BAZ]"
+          + "}",
+        builder);
+
+        assertEquals(0, builder.getRepeatedNestedEnumValue(0));
```

Comment:
Same here. I think the entry should be discarded instead of converted to a different value.